### PR TITLE
Feat/15085/banner base component

### LINKED ISSUE #15085: [Ext Nav] Create component: BannerBase
### Description
Create a reusable UI component for AvatarWithToken

### References
[Figma Link](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/Design-System-2.0?node-id=1915%3A12258)
UI component guidelines: TBC
CSS guidelines: TBC
Testing guidelines: TBC

### Files needed
- `index.js`
- `index.scss`
- `BannerBase.js`
- `BannerBase.stories.js`
- `BannerBase.test.js`
- `README.mdx`

### Technical details
TBC

### Acceptance criteria
- [ ] Uses color, typography, shadows design tokens 
- [ ] Uses semantic html
- [ ] PropTypes have descriptions
- [ ] Has storybook story with controls
- [ ] Has documentation in MDX
- [ ] Has unit tests and 90% coverage
- [ ] Works in Chrome and Firefox
- [ ] Performance tested: no unnecessary re-renders or other performance concerns

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation
* Fixes #15085 

### Before

n/a

### After

Screenshot 2022-12-07 at 1 46 47 AM

## Manual Testing Steps

`yarn test:unit:jest .ui/components/component-library/banner-base/banner-base.test.js `

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by NidhiKJha — COMMENTED** [review]
Looking great 🚀. Just a couple of suggestions

File: ui/components/component-library/banner-base/banner-base.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        {title && (
+          <Text className="mm-banner-base__title" variant={TEXT.BODY_LG_MEDIUM}>
+            {title}
+          </Text>
+        )}
+        {description && (
+          <Text className="mm-banner-base__description">{description}</Text>
+        )}
+        {action && <Text className="mm-banner-base__action">{action}</Text>}
+      </div>
```
Comment (NidhiKJha) [inline ui/components/component-library/banner-base/banner-base.js:49]: Can we use fragment here?
```suggestion

      <>
        {title && (
          <Text className="mm-banner-base__title" variant={TEXT.BODY_LG_MEDIUM}>
            {title}
          </Text>
        )}
        {description && (
          <Text className="mm-banner-base__description">{description}</Text>
        )}
        {action && <Text className="mm-banner-base__action">{action}</Text>}
      </>
```

File: ui/components/component-library/banner-base/banner-base.stories.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      control: 'select',
+      table: { category: 'box props' },
+    },
+    marginLeft: {
+      options: marginSizeControlOptions,
+      control: 'select',
+      table: { category: 'box props' },
+    },
+  },
+};
```
Comment (NidhiKJha) [inline ui/components/component-library/banner-base/banner-base.stories.js:61]:
```suggestion
  argTypes: {
    className: {
      control: 'text',
    },
    title: {
      control: 'text',
    },
    description: {
      control: 'text',
    },
    action: {
      control: 'func',
    },
    leftAccessory: {
      control: 'text',
    },
    onClose: {
      action: 'onClose',
    },
    marginTop: {
      options: marginSizeControlOptions,
      control: 'select',
      table: { category: 'box props' },
    },
    marginRight: {
      options: marginSizeControlOptions,
      control: 'select',
      table: { category: 'box props' },
    },
    marginBottom: {
      options: marginSizeControlOptions,
      control: 'select',
      table: { category: 'box props' },
    },
    marginLeft: {
      options: marginSizeControlOptions,
      control: 'select',
      table: { category: 'box props' },
    },
  },
};
```

File: ui/components/component-library/banner-base/README.mdx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+<BannerBase
+  title="Action prop demo"
+  description="Call to action items will appear below this description"
+  action={<ButtonLink size={SIZES.AUTO}>Action</ButtonLink>}
+/>;
+```
+
+### On Close
+
+Use the `onClick`` prop to pass a function to the close button. The close button will appear when this prop is used.
```
Comment (NidhiKJha) [inline ui/components/component-library/banner-base/README.mdx:81]:
```suggestion
Use the `onClose` prop to pass a function to the close button. The close button will appear when this prop is used.
```

**Review by NidhiKJha — COMMENTED** [review]
Looking really good 🚀 . Just a small nit

**Review by georgewrmarshall — COMMENTED** [review]
Looking good overall! I think we could make it a little more extensible and allow for custom content? Left some comments and some questions. Also should the title, description and action be stacked?

Screenshot 2022-12-09 at 5 53 37 PM

File: ui/components/component-library/banner-base/banner-base.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+      <>
+        {title && (
+          <Text className="mm-banner-base__title" variant={TEXT.BODY_LG_MEDIUM}>
+            {title}
+          </Text>
+        )}
+        {description && (
+          <Text className="mm-banner-base__description">{description}</Text>
+        )}
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/banner-base/banner-base.js:49]: material example https://mui.com/material-ui/react-alert/

**Review by Seifollahy — APPROVED** [review]

**Review by Seifollahy — DISMISSED** [review]

**Comment by georgewrmarshall** [comment]
We should also be thinking can this replace what already exists. Could an engineer use this component to create this component ui/components/app/home-notification/home-notification.component.js

File: ui/components/component-library/banner-base/banner-base.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      className={classnames('mm-banner-base', className)}
+      display={DISPLAY.INLINE_FLEX}
+      gap={2}
+      backgroundColor={COLORS.BACKGROUND_DEFAULT}
+      borderRadius={SIZES.SM}
+      padding={3}
+      {...props}
+    >
+      {leftAccessory && (
+        <div className="mm-banner-base__left-accessory">{leftAccessory}</div>
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/banner-base/banner-base.js:39]: Seems like html bloat? Does the flex styles distort the icon if this is removed?

File: ui/components/component-library/banner-base/banner-base.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      <>
+        {title && (
+          <Text className="mm-banner-base__title" variant={TEXT.BODY_LG_MEDIUM}>
+            {title}
+          </Text>
+        )}
+        {description && (
+          <Text className="mm-banner-base__description">{description}</Text>
+        )}
+        {action && <Text className="mm-banner-base__action">{action}</Text>}
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/banner-base/banner-base.js:55]: Yeah that's also an option. We could just provide it as an example of how actions should be displayed

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! One thing that @darkwing raised in a DM yesterday is that if we want to be more inclusive and support RTL better we should avoid `left` and `right` in favour of `start` and `end`. Could be a separate effort but something to keep in mind

File: ui/components/component-library/banner-base/banner-base.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { ButtonIcon } from '../button-icon';
+import { ButtonLink } from '../button-link';
+import { ICON_NAMES } from '../icon';
+import { Text } from '../text';
```
Comment (NidhiKJha) [inline ui/components/component-library/banner-base/banner-base.js:8]:
```suggestion
import { ButtonIcon, ButtonLink,  ICON_NAMES,  Text} from '..';

```

**Review by NidhiKJha — COMMENTED** [review]
LGTM! The import fix is really good 🔥 Just a couple of suggestions

File: ui/components/component-library/banner-base/README.mdx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+## Props
+
+The `BannerBase` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) component props
+
+<ArgsTable of={BannerBase} />
+
+### Title
+
+Use the `title` prop to pass a string that is sentence case no period
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-base/README.mdx:22]: Could mention `titleProps` here
```suggestion
Use the `title` prop to pass a string that is sentence case no period. Use the `titleProps` prop to pass additional props to the `Text` component.
```

File: ui/components/component-library/banner-base/banner-base.js [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+          </Text>
+        )}
+        {children && (
+          <Text
+            as={typeof children === 'object' ? 'div' : 'p'}
+            className="mm-banner-base__description"
+          >
+            {children}
+          </Text>
+        )}
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-base/banner-base.js:62]: Ooh ok 🧠 this is interesting! Could we take this one step further and remove the nesting of a div if `children` are nodes?

```suggestion
        {children && typeof children === 'object' ? (
          children
        ) : (
          <Text>{children}</Text>
        )}
```

File: ui/components/component-library/banner-base/banner-base.js [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+        {actionButtonLabel && (
+          <ButtonLink
+            // noPadding TODO: Use noPadding option when released
+            size={SIZES.AUTO} // TODO: Remove when noPadding is added
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-base/banner-base.js:66]: Oh nice! I found another reason `SIZES.AUTO` doesn't quite work for stand alone ButtonLinks in the main comment of this review

File: ui/components/component-library/banner-base/banner-base.js [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+  );
+};
+
+BannerBase.propTypes = {
+  /**
+   * The title of the BannerBase
+   */
+  title: PropTypes.string,
+  /**
+   * The title of the BannerBase
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-base/banner-base.js:94]: Update prop type description
```suggestion
   * Additional props to pass to the `Text` component used for the `title` text
```

File: ui/components/component-library/banner-base/banner-base.test.js [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+    );
+  });
+
+  it('should render bannerbase action button', () => {
+    const { getByTestId } = render(
+      <BannerBase
+        title="Action prop demo"
+        actionButtonLabel="Action"
+        actionButtonProps={{
+          icon: ICON_NAMES.ARROW_2_RIGHT, // TODO: change to iconName
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-base/banner-base.test.js:56]: Can this be updated now?
```suggestion
          iconName: ICON_NAMES.ARROW_2_RIGHT,
```

File: ui/components/component-library/banner-base/banner-base.test.js [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+          />
+        }
+      />,
+    );
+
+    const icon = container.getElementsByClassName(
+      'mm-banner-base__start-accessory',
+    ).length;
+    expect(icon).toBe(1);
+  });
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-base/banner-base.test.js:89]: Optional. Could use data-testid here for brevity

```suggestion
  it('should render bannerbase startAccessory', () => {
    const { container, getByTestId } = render(
      <BannerBase
        startAccessory={
          <Icon
            data-testid="start-accessory"
            name={ICON_NAMES.ADD_SQUARE_FILLED}
          />
        }
      />,
    );

    expect(getByTestId('start-accessory').toBeDefined();
  });
```

File: ui/components/component-library/index.js [hunk 14]
```diff
@@ -25,3 +27,4 @@ export {
   TEXT_FIELD_BASE_TYPES,
 } from './text-field-base';
 export { TextFieldSearch } from './text-field-search';
+export { BannerBase } from './banner-base';
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/index.js:31]: This is sooo weird someone please explain to my smooth brain

**Review by georgewrmarshall — COMMENTED** [review]
Tiiight 👌 Couple small suggestions I missed in last review

Another issue with ButtonLink size.inherit for buttons instead of inline links is that the font-size doesn't adhere to the small screen / large screen changes. So on small screens it looks too big


https://user-images.githubusercontent.com/8112138/209027904-5480582b-cc45-454f-8b0a-18cf71ffefc7.mov

**Review by NidhiKJha — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🔥 Some non-blocking nice to have updates
- Add action controls for close and action buttons in storybook
- Getting a error on startAccessory controls

Screenshot 2023-01-09 at 12 03 36 PM

**Pull request merged**